lib/Driver/ToolChains/Gnu.h

```diff
@@ -62,7 +65,8 @@ class LLVM_LIBRARY_VISIBILITY Assembler : public GnuTool {
 };
 
 class LLVM_LIBRARY_VISIBILITY Linker : public GnuTool {
-  mutable std::unique_ptr<Linker> HCLinker;
+  mutable std::unique_ptr<Tool> HCLinker;
+  friend class HCC::CXXAMPLink;
```

Comment:
I believe the reason for this friend class is allow `CXXAMPLink` to call the `ConstructLinkerJob`?  If that's the case, is it really necessary to call `gnutools::Linker::ConstructLinkerJob` from `CXXAMPLink::ConstructJob`?
In other words, could `CXXAMPLink` just return the additional arg back to `gnutools::Linker`  for it to call its own `ConstructLinkerJob`?